src/client.rs

```diff
@@ -641,6 +643,30 @@ impl<C: Config> Client<C> {
         self.bits.hardcore()
     }
 
+    /// Requests that the client download and enable a resource pack.
+    ///
+    /// # Arguments
+    /// * `url` - The URL of the resource pack file.
+    /// * `hash` - The SHA-1 hash of the resource pack file.
+    /// * `forced` - Whether a client should be kicked from the server upon
+    ///   declining the pack (this is enforced client-side)
+    /// * `prompt_message` - A message to be displayed with the resource pack
+    ///   dialog.
+    pub fn set_resource_pack(
+        &mut self,
+        url: impl Into<String>,
+        hash: impl Into<Option<String>>,
```

Comment:
I think it would be best to make this a required argument since it is required by the packet. It would also be good to describe the constraints on the value of the string.

Reply:
The reason I didn't make it required is because hash checking on server resource packs is optional, and will be skipped if the server sends a non-hash string (like ""). So we would need to explicitly state in the doc that to skip hash checking you need to pass an empty string in (which I felt like `Option` would be more suited for in this case)

Reply:
The use of `Option` here implies a semantic difference between `None` and `Some("")`, but there is not. There isn't any additional type safety gained by adding an Option. If we really wanted to we could use more precise types but that isn't really worth the effort.

Reply:
Yeah that's true, I can see how people might get confused over `None` vs `Some("")`

Reply:
> (except that the string should be lowercase).
Hex strings are case-insensitive and I checked that an uppercase hash works, so we can probably leave this out.

Reply:
Yeah I guess the wiki.vg info is outdated. I will have to update it.